transactions/transaction.go

```diff
@@ -149,6 +160,7 @@ type Manager interface {
 	// ExecuteWithinTX executes the given unit of work within a transaction
 	ExecuteWithinTX(ctx context.Context, accountID identity.DID, existingTxID TxID, desc string, work func(accountID identity.DID, txID TxID, txMan Manager, err chan<- error)) (txID TxID, done chan bool, err error)
 	GetTransaction(accountID identity.DID, id TxID) (*Transaction, error)
+	UpdateTransactionWithValue(accountID identity.DID, id TxID, key string, value []byte) error
```

Comment:
do we need to have this as part of the interface?

Reply:
I would like to, as this is the base for allowing the storage of values that are calculated in an ansynchronous task that can be consumed later on.

Reply:
I understand, but the tasks are only allowed to update transactions using base task, transactions should not be updated unless within transactions package, because it could cause race conditions and other weird side effects. So we should not expose any update methods on the transaction manager.